Source/Scene/modelMaterialsCommon.js

```diff
@@ -0,0 +1,792 @@
+                        typeValue = WebGLConstants.SAMPLER_2D;
+                        break;
+                    case 'number':
+                        typeValue = WebGLConstants.FLOAT;
+                        if (!hasAlpha && name === 'transparency') {
+                            hasAlpha = (value !== 1.0);
+                        }
+                        break;
+                    default:
+                        if (Array.isArray(value)) {
+                            // 35664 (vec2), 35665 (vec3), 35666 (vec4)
+                            typeValue = 35662 + value.length;
+                            if (!hasAlpha && typeValue === WebGLConstants.FLOAT_VEC4 && name === 'diffuse') {
+                                hasAlpha = (value[3] !== 1.0);
+                            }
+                        }
+                        break;
+                }
+                if (typeValue > 0) {
+                    lowerCase = name.toLowerCase();
+                    techniqueParameters[lowerCase] = {
+                        type: typeValue
+                    };
+                }
+            }
+        }
+
+        // Copy light parameters into technique parameters
+        if (defined(lightParameters)) {
+            for (var lightParamName in lightParameters) {
+                if (lightParameters.hasOwnProperty(lightParamName)) {
+                    techniqueParameters[lightParamName] = lightParameters[lightParamName];
+                }
+            }
+        }
+
+        // Generate uniforms object before attributes are added
+        var techniqueUniforms = {};
+        for (var paramName in techniqueParameters) {
+            if (techniqueParameters.hasOwnProperty(paramName)) {
+                var param = techniqueParameters[paramName];
+                techniqueUniforms['u_' + paramName] = paramName;
+                var arraySize = defined(param.count) ? '['+param.count+']' : '';
+                if (param.type !== WebGLConstants.FLOAT_MAT3 && param.type !== WebGLConstants.FLOAT_MAT4) {
+                    fragmentShader += 'uniform ' + webGLConstantToGlslType(param.type) + ' u_' + paramName + arraySize + ';\n';
+                }
+                else {
+                    vertexShader += 'uniform ' + webGLConstantToGlslType(param.type) + ' u_' + paramName + arraySize + ';\n';
+                }
+            }
+        }
+
+        // Add attributes with semantics
+        var vertexShaderMain = '';
+        if (hasSkinning) {
+            vertexShaderMain += '  mat4 skinMat = a_weight.x * u_jointMatrix[int(a_joint.x)];\n';
+            vertexShaderMain += '  skinMat += a_weight.y * u_jointMatrix[int(a_joint.y)];\n';
+            vertexShaderMain += '  skinMat += a_weight.z * u_jointMatrix[int(a_joint.z)];\n';
+            vertexShaderMain += '  skinMat += a_weight.w * u_jointMatrix[int(a_joint.w)];\n';
+        }
+
+        // TODO: Handle multiple texture coordinates for now we just use 1
```

Comment:
If we are OK with the spec just supporting texture coordinates 0 (I am), then remove this TODO.